### ISSUE TITLE
`@layer` with properties causes parsing error

### ISSUE THREAD

**Issue by xbinaryx** [eslint/csstree#106]
### What did you do?

```css
.foo { @layer base { color: red; } }
```

### What did you expect to happen?

No errors reported

### What actually happened?

Parser throws "Identifier is expected" when parsing nested `@layer` rules that contain properties, which is valid CSS according to the [CSS Nesting](https://www.w3.org/TR/css-nesting-1/#conditionals) specification.

### Link to Minimal Reproducible Example

[Code Explorer](https://explorer.eslint.org/#eslint-explorer=IntcInN0YXRlXCI6e1widG9vbFwiOlwiYXN0XCIsXCJjb2RlXCI6e1wiamF2YXNjcmlwdFwiOlwiLyoqXFxuICogVHlwZSBvciBwYXN0ZSBzb21lIEphdmFTY3JpcHQgaGVyZSB0byBsZWFybiBtb3JlIGFib3V0XFxuICogdGhlIHN0YXRpYyBhbmFseXNpcyB0aGF0IEVTTGludCBjYW4gZG8gZm9yIHlvdS5cXG4gKiBcXG4gKiBUaGUgdGhyZWUgdGFicyBhcmU6XFxuICogXFxuICogLSBBU1QgLSBUaGUgQWJzdHJhY3QgU3ludGF4IFRyZWUgb2YgdGhlIGNvZGUsIHdoaWNoIGNhblxcbiAqICAgYmUgdXNlZnVsIHRvIHVuZGVyc3RhbmQgdGhlIHN0cnVjdHVyZSBvZiB0aGUgY29kZS4gWW91XFxuICogICBjYW4gdmlldyB0aGlzIHN0cnVjdHVyZSBhcyBKU09OIG9yIGluIGEgdHJlZSBmb3JtYXQuXFxuICogLSBTY29wZSAtIFRoZSBzY29wZSBzdHJ1Y3R1cmUgb2YgdGhlIGNvZGUsIHdoaWNoIGNhbiBiZVxcbiAqICAgdXNlZnVsIHRvIHVuZGVyc3RhbmQgaG93IHZhcmlhYmxlcyBhcmUgZGVmaW5lZCBhbmRcXG4gKiAgIHdoZXJlIHRoZXkgYXJlIHVzZWQuXFxuICogLSBDb2RlIFBhdGggLSBUaGUgY29kZSBwYXRoIHN0cnVjdHVyZSBvZiB0aGUgY29kZSwgd2hpY2hcXG4gKiAgIGNhbiBiZSB1c2VmdWwgdG8gdW5kZXJzdGFuZCBob3cgdGhlIGNvZGUgaXMgZXhlY3V0ZWQuXFxuICogXFxuICogWW91IGNhbiBjaGFuZ2UgdGhlIHdheSB0aGF0IHRoZSBKYXZhU2NyaXB0IGNvZGUgaXMgaW50ZXJwcmV0ZWRcXG4gKiBieSBjbGlja2luZyBcXFwiSmF2YVNjcmlwdFxcXCIgaW4gdGhlIGhlYWRlciBhbmQgc2VsZWN0aW5nIGRpZmZlcmVudFxcbiAqIG9wdGlvbnMuXFxuICovXFxuXFxuaW1wb3J0IGpzIGZyb20gXFxcIkBlc2xpbnQvanNcXFwiO1xcbmltcG9ydCB7IGRlZmluZUNvbmZpZyB9IGZyb20gXFxcImVzbGludC9jb25maWdcXFwiO1xcblxcbmZ1bmN0aW9uIGdldENvbmZpZygpIHtcXG4gICAgcmV0dXJuIHtcXG4gICAgICAgIHJ1bGVzOiB7XFxuICAgICAgICAgICAgXFxcInByZWZlci1jb25zdFxcXCI6IFxcXCJ3YXJuXFxcIlxcbiAgICAgICAgfVxcbiAgICB9O1xcbn1cXG5cXG5leHBvcnQgZGVmYXVsdCBkZWZpbmVDb25maWcoW1xcbiAgICBqcy5jb25maWdzLnJlY29tbWVuZGVkLFxcbiAgICBnZXRDb25maWcoKVxcbl0pO1wiLFwianNvblwiOlwiLyoqXFxuICogVHlwZSBvciBwYXN0ZSBzb21lIEpTT04gaGVyZSB0byBsZWFybiBtb3JlIGFib3V0XFxuICogdGhlIHN0YXRpYyBhbmFseXNpcyB0aGF0IEVTTGludCBjYW4gZG8gZm9yIHlvdS5cXG4gKlxcbiAqIFRoZSB0YWJzIGFyZTpcXG4gKlxcbiAqIC0gQVNUIC0gVGhlIEFic3RyYWN0IFN5bnRheCBUcmVlIG9mIHRoZSBjb2RlLCB3aGljaCBjYW5cXG4gKiAgIGJlIHVzZWZ1bCB0byB1bmRlcnN0YW5kIHRoZSBzdHJ1Y3R1cmUgb2YgdGhlIGNvZGUuIFlvdVxcbiAqICAgY2FuIHZpZXcgdGhpcyBzdHJ1Y3R1cmUgYXMgSlNPTiBvciBpbiBhIHRyZWUgZm9ybWF0LlxcbiAqXFxuICogWW91IGNhbiBjaGFuZ2UgdGhlIHdheSB0aGF0IHRoZSBKU09OIGNvZGUgaXMgaW50ZXJwcmV0ZWRcXG4gKiBieSBjbGlja2luZyBcXFwiSlNPTlxcXCIgaW4gdGhlIGhlYWRlciBhbmQgc2VsZWN0aW5nIGRpZmZlcmVudFxcbiAqIG9wdGlvbnMuXFxuICpcXG4gKiBUaGlzIGV4YW1wbGUgaXMgaW4gSlNPTkMgbW9kZSwgd2hpY2ggYWxsb3dzIGNvbW1lbnRzLlxcbiAqL1xcblxcbntcXG4gICAgXFxcImtleTFcXFwiOiBbdHJ1ZSwgZmFsc2UsIG51bGxdLFxcbiAgICBcXFwia2V5MlxcXCI6IHtcXG4gICAgICAgIFxcXCJrZXkzXFxcIjogWzEsIDIsIFxcXCIzXFxcIiwgMWUxMCwgMWUtM11cXG4gICAgfVxcbn1cIixcIm1hcmtkb3duXCI6XCI8IS0tXFxuVHlwZSBvciBwYXN0ZSBzb21lIE1hcmtkb3duIGhlcmUgdG8gbGVhcm4gbW9yZSBhYm91dFxcbnRoZSBzdGF0aWMgYW5hbHlzaXMgdGhhdCBFU0xpbnQgY2FuIGRvIGZvciB5b3UuXFxuXFxuVGhlIHRhYnMgYXJlOlxcblxcbi0gQVNUIC0gVGhlIEFic3RyYWN0IFN5bnRheCBUcmVlIG9mIHRoZSBjb2RlLCB3aGljaCBjYW5cXG5iZSB1c2VmdWwgdG8gdW5kZXJzdGFuZCB0aGUgc3RydWN0dXJlIG9mIHRoZSBjb2RlLiBZb3VcXG5jYW4gdmlldyB0aGlzIHN0cnVjdHVyZSBhcyBKU09OIG9yIGluIGEgdHJlZSBmb3JtYXQuXFxuXFxuWW91IGNhbiBjaGFuZ2UgdGhlIHdheSB0aGF0IHRoZSBNYXJrZG93biBjb2RlIGlzIGludGVycHJldGVkXFxuYnkgY2xpY2tpbmcgXFxcIk1hcmtkb3duXFxcIiBpbiB0aGUgaGVhZGVyIGFuZCBzZWxlY3RpbmcgZGlmZmVyZW50XFxub3B0aW9ucy5cXG5cXG5UaGlzIGV4YW1wbGUgaXMgaW4gQ29tbW9uTWFyayBtb2RlLlxcbi0tPlxcblxcbiMgRVNMaW50IE1hcmtkb3duIEV4YW1wbGVcXG5cXG5UaGlzIGlzIGFuIGV4YW1wbGUgb2YgYSBNYXJrZG93biBmaWxlIHRoYXQgY2FuIGJlIHBhcnNlZFxcbmJ5IEVTTGludC4gTWFya2Rvd24gaXMgYSBzaW1wbGUgbWFya3VwIGxhbmd1YWdlIHRoYXQgaXNcXG5vZnRlbiB1c2VkIGZvciBkb2N1bWVudGF0aW9uLlxcblxcbiMjIEZlYXR1cmVzXFxuXFxuLSBNYWtlIHRoaW5ncyAqaXRhbGljKiwgKipib2xkKiosIG9yIGBjb2RlYFxcbi0gQ3JlYXRlIFtsaW5rc10oaHR0cHM6Ly9lc2xpbnQub3JnKVxcbi0gU3VwcG9ydHMgSFRNTCA8c3BhbiBzdHlsZT1cXFwiY29sb3I6IHJlZDtcXFwiPmVsZW1lbnRzPC9zcGFuPlxcbi0gTGlzdHNcXG4gIC0gTmVzdGVkIGxpc3RzXCIsXCJjc3NcIjpcIi5mb28geyBAbGF5ZXIgYmFzZSB7IGNvbG9yOiByZWQ7IH0gfVwiLFwiaHRtbFwiOlwiPCFET0NUWVBFIGh0bWw%2BXFxuPCEtLVxcblR5cGUgb3IgcGFzdGUgc29tZSBIVE1MIGhlcmUgdG8gbGVhcm4gbW9yZSBhYm91dFxcbnRoZSBzdGF0aWMgYW5hbHlzaXMgdGhhdCBFU0xpbnQgY2FuIGRvIGZvciB5b3UuXFxuXFxuVGhlIHRhYnMgYXJlOlxcblxcbi0gQVNUIC0gVGhlIEFic3RyYWN0IFN5bnRheCBUcmVlIG9mIHRoZSBjb2RlLCB3aGljaCBjYW5cXG5iZSB1c2VmdWwgdG8gdW5kZXJzdGFuZCB0aGUgc3RydWN0dXJlIG9mIHRoZSBjb2RlLiBZb3VcXG5jYW4gdmlldyB0aGlzIHN0cnVjdHVyZSBhcyBKU09OIG9yIGluIGEgdHJlZSBmb3JtYXQuXFxuLS0%2BXFxuXFxuPGh0bWwgbGFuZz1cXFwiZW5cXFwiPlxcbiAgICA8aGVhZD5cXG4gICAgICAgIDxtZXRhIGNoYXJzZXQ9XFxcIlVURi04XFxcIj5cXG4gICAgICAgIDx0aXRsZT5IVE1MPC90aXRsZT5cXG4gICAgPC9oZWFkPlxcbiAgICA8Ym9keT5cXG4gICAgICAgIDxwPlRleHQ8L3A%2BXFxuICAgIDwvYm9keT5cXG48L2h0bWw%2BXCJ9LFwibGFuZ3VhZ2VcIjpcImNzc1wiLFwianNPcHRpb25zXCI6e1wicGFyc2VyXCI6XCJlc3ByZWVcIixcInNvdXJjZVR5cGVcIjpcIm1vZHVsZVwiLFwiZXNWZXJzaW9uXCI6XCJsYXRlc3RcIixcImlzSlNYXCI6dHJ1ZX0sXCJqc29uT3B0aW9uc1wiOntcImpzb25Nb2RlXCI6XCJqc29uY1wiLFwiYWxsb3dUcmFpbGluZ0NvbW1hc1wiOmZhbHNlfSxcImNzc09wdGlvbnNcIjp7XCJjc3NNb2RlXCI6XCJjc3NcIixcInRvbGVyYW50XCI6ZmFsc2V9LFwibWFya2Rvd25PcHRpb25zXCI6e1wibWFya2Rvd25Nb2RlXCI6XCJjb21tb25tYXJrXCIsXCJtYXJrZG93bkZyb250bWF0dGVyXCI6XCJvZmZcIn0sXCJodG1sT3B0aW9uc1wiOntcInRlbXBsYXRlRW5naW5lU3ludGF4XCI6XCJub25lXCIsXCJmcm9udG1hdHRlclwiOmZhbHNlfSxcIndyYXBcIjp0cnVlLFwidmlld01vZGVzXCI6e1wiYXN0Vmlld1wiOlwidHJlZVwiLFwic2NvcGVWaWV3XCI6XCJmbGF0XCIsXCJwYXRoVmlld1wiOlwiZ3JhcGhcIn0sXCJwYXRoSW5kZXhcIjp7XCJpbmRleFwiOjAsXCJpbmRleGVzXCI6MX0sXCJlc3F1ZXJ5U2VsZWN0b3JcIjp7XCJzZWxlY3RvclwiOlwiXCJ9fSxcInZlcnNpb25cIjowfSI%3D)

### Participation

- [x] I am willing to submit a pull request for this issue.
